src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VisualBasic/SplashScreenEnumProvider.cs

```diff
@@ -0,0 +1,115 @@
+        // remove this and always include the "(Not set)" value.
+        bool includeEmptyValue = options?.Any(pair =>
+            pair.Name == "IncludeEmptyValue"
+            && bool.TryParse(pair.Value, out bool optionValue)
+            && optionValue) ?? false;
+
+        return Task.FromResult<IDynamicEnumValuesGenerator>(new SplashScreenEnumGenerator(_workspace, _unconfiguredProject, _properties, includeEmptyValue, true));
+    }
+
+    internal class SplashScreenEnumGenerator : IDynamicEnumValuesGenerator
+    {
+        public bool AllowCustomValues => false;
+        private readonly Workspace _workspace;
+        private readonly UnconfiguredProject _unconfiguredProject;
+        private readonly ProjectProperties _properties;
+        private readonly bool _includeEmptyValue;
+        private readonly bool _searchForEntryPointsInFormsOnly;
+
+        public SplashScreenEnumGenerator(Workspace workspace, UnconfiguredProject unconfiguredProject, ProjectProperties properties, bool includeEmptyValue, bool searchForEntryPointsInFormsOnly)
+        {
+            _workspace = workspace;
+            _unconfiguredProject = unconfiguredProject;
+            _properties = properties;
+            _includeEmptyValue = includeEmptyValue;
+            _searchForEntryPointsInFormsOnly = searchForEntryPointsInFormsOnly;
+        }
+
+        public async Task<ICollection<IEnumValue>> GetListedValuesAsync()
+        {
+            Project? project = _workspace.CurrentSolution.Projects.FirstOrDefault(p => PathHelper.IsSamePath(p.FilePath!, _unconfiguredProject.FullPath));
+            if (project is null)
+            {
+                return Array.Empty<IEnumValue>();
+            }
+
+            Compilation? compilation = await project.GetCompilationAsync();
+            if (compilation is null)
+            {
+                // Project does not support compilations
+                return Array.Empty<IEnumValue>();
+            }
+
+            List<IEnumValue> enumValues = new();
+            if (_includeEmptyValue)
+            {
+                enumValues.Add(new PageEnumValue(new EnumValue { Name = "", DisplayName = VSResources.StartupObjectNotSet }));
+            }
+
+            IEntryPointFinderService? entryPointFinderService = project.Services.GetService<IEntryPointFinderService>();
+            IEnumerable<INamedTypeSymbol>? entryPoints = entryPointFinderService?.FindEntryPoints(compilation.GlobalNamespace, _searchForEntryPointsInFormsOnly);
+            if (entryPoints is not null)
+            {
+                enumValues.AddRange(entryPoints.Select(ep =>
+                {
+                    string displayName = ep.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted));
+                    return new PageEnumValue(new EnumValue { Name = ep.Name, DisplayName = displayName });
```

Comment:
It'd be good to document why both name and full name are used here, including what's written in the `.myapp` file.

Also what happens when the user types in a value that's not in the list. My guess is it'd end up in the `.myapp` file with the root namespace prefixed.

Another question is what happens if the `.myapp` file contains a fully qualified name? Does it find the splash screen? If so, would we fail to select it in the UI here, or would we create two entries with the same `DisplayName` but different `Name` values?

Reply:
A user can enter a value in the myapp file directly and we will try to show it in the UI as it is, using the `TryCreateEnumValueAsync` method below. So, we will not be adding the root namespace prefix, just have the value be the name and display name for that enum value.

As for the question around having the fully qualified name, the code won't be generated then and we would create two different entries in the UI, the one we picked up and this fully qualified one (we would generate it via the `TryCreateEnumValueAsync`).